File: Practica Auxi/POO/Ejercicio 3/Python/Coche.py
#3. Crea una clase Coche con marca, modelo y velocidad
#a) Agrega un método acelerar () que aumente la velocidad en 10
#b) Agrega un método frenar () que disminuya la velocidad en 5
#c) Crea dos coches, aceléralos, frénalos y muestra sus velocidades
class Coche:
    def __init__(self, marca, modelo, velocidad):
        self.marca = marca
        self.modelo = modelo
        self.velocidad = velocidad

    def acelerar(self):
        print(self)  
        print(f"El coche aceleró de {self.velocidad} km/h")
        self.velocidad += 10
        print(f"Velocidad actual: {self.velocidad} km/h")

    def  frenar(self):
        self.velocidad -= 5
        print(self)
        print(f"El coche freno de {self.velocidad} km/h")
        self.velocidad -= 5
        if self.velocidad < 0:  
            self.velocidad = 0
        print(f"Velocidad actual: {self.velocidad} km/h")

    def __str__(self):
        return f"Marca: {self.marca}, Modelo: {self.modelo}, Velocidad: {self.velocidad} km/h"

Coche1 = Coche("Nissan", "Coupe", 87)
Coche1.acelerar()

Coche2 = Coche("Toyota", "Deportivo", 120) 
Coche2.frenar()

File: Practica Auxi/POO/Ejercicio 3/Java/Coche.java
/*
3. Crea una clase Coche con marca, modelo y velocidad
   a) Agrega un método acelerar() que aumente la velocidad en 10
   b) Agrega un método frenar() que disminuya la velocidad en 5
   c) Crea dos coches, aceléralos, frénalos y muestra sus velocidades
*/
public class Coche {
    private String marca;
    private String modelo;
    private int velocidad;

    public Coche(String marca, String modelo, int velocidad) {
        this.marca = marca;
        this.modelo = modelo;
        this.velocidad = velocidad;
    }

    public void acelerar() {
        System.out.println(this);  
        System.out.println("El coche aceleró de " + velocidad + " km/h");
        velocidad += 10;
        System.out.println("Velocidad actual: " + velocidad + " km/h");
    }

    public void frenar() {
        System.out.println(this);
        System.out.println("El coche frenó de " + velocidad + " km/h");
        velocidad -= 5;
        if (velocidad < 0) {
            velocidad = 0;
        }
        System.out.println("Velocidad actual: " + velocidad + " km/h");
    }

    @Override
    public String toString() {
        return "Marca: " + marca + ", Modelo: " + modelo + ", Velocidad: " + velocidad + " km/h";
    }
}